rxfingerprint/src/main/java/com/mtramin/rxfingerprint/FingerprintObservable.java

```diff
@@ -115,6 +117,20 @@ public void onAuthenticationSucceeded(FingerprintManagerCompat.AuthenticationRes
 	}
 
 	/**
+	 * Required to catch errors from certain phones. See:
+	 * http://stackoverflow.com/questions/36043912/error-after-fingerprint-touched-on-samsung-phones-android-security-keystoreexce
+	 * @param e Exception caught
+	 * @param subscriber Subscriber for fingerprint result
+     */
+	void checkKey(Exception e, Subscriber<? extends FingerprintAuthenticationResult> subscriber, Action0 action0) {
```

Comment:
Well, the issue is that the `KeyPermanentlyInvalidatedException` can (on some popular devices) manifest itself as a `android.security.KeyStoreException`. So, while it may not seem like the same issue, if you read the referenced stackoverflow link, it is caused by a fingerprint being added while the app is in the background.

So, I felt it was necessary to resolve both rather than just one. :)